keyless/pepper/README.md

```diff
@@ -0,0 +1,110 @@
+# Pepper Service Developer Guide
+
+## Requirements
+
+The Pepper Service requires an account recovery database, which is deployed as a Firestore instance on GCP.
+If you wish to run the Pepper Service locally, you will need to set up a Firestore emulator. This requires:
+1. **GCP Project**: You must have a GCP project already set up.
+2. **Service Account**: You need to create a service account with Firestore access and download the credential file 
+   in JSON format.
+
+Both of these steps can be done via the GCP console. Once you have completed these steps, you can proceed.
+
+## Start the Firestore Emulator
+
+First, you will need to start the Firestore emulator locally. This can be done by running the
+`start-firestore-emulator.sh` script in this repository.
+
+```bash
+cd keyless/pepper/scripts
+./start-firestore-emulator.sh
+````
+
+Note that the Firestore emulator will run in the foreground, so you will need to keep this terminal
+window open while you work with the Pepper Service.
+
+### Stop the Emulator
+
+If you ever need to stop the Firestore emulator, you can do so by running the `stop-firestore-emulator.sh` script:
+```bash
+cd keyless/pepper/scripts
+./stop-firestore-emulator.sh
+```
+
+## Start the Pepper Service
+
+Next, you can start the Pepper Service by running the `start-pepper-service.sh` script in this repository.
+
+```bash
+cd keyless/pepper/scripts
+./start-pepper-service.sh <FIRESTORE_EMULATOR_HOST> <GOOGLE_APPLICATION_CREDENTIALS> <PROJECT_ID>
+```
+
+The script arguments are:
+- `<FIRESTORE_EMULATOR_HOST>`: The host and port of the Firestore emulator, e.g., `localhost:8081`.
+- `<GOOGLE_APPLICATION_CREDENTIALS>`: The path to your service account credential file (in JSON format).
+- `<PROJECT_ID>`: The ID of your GCP project.
+
+Note that the Pepper Service will run in the foreground, so you will need to keep this terminal
+window open while you interact with the Pepper Service.
+
+## Run the Pepper Client Example
+
+Finally, you can run the example client to interact with the Pepper Service. The client will send a
+pepper request and verify the response. It also connects to the account recovery database and
+verifies that it was correctly updated by the Pepper Service.
+
+To run the example client, execute the `start-pepper-client.sh` script in this repository.
+
+```bash
+cd keyless/pepper/scripts
+./start-pepper-client.sh <FIRESTORE_EMULATOR_HOST> <GOOGLE_APPLICATION_CREDENTIALS> <PROJECT_ID>
+```
+
+The script arguments are the same as those used for starting the Pepper Service (above).
+
+Note: the client is an interactive console program, so you will need to follow the instructions
```

Comment:
Is the interaction necessary for some reason? e.g., like pasting in a JWT?

Reply:
Yes, exactly. The biggest things it requires are: (i) prompts to create the JWT; and (ii) prompts to enter the JWT. We can think about simplifying it, but to be honest, it's actually not too bad (especially if you're learning about the flow 😄)